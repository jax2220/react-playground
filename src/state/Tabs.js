import React from 'react';

class Tabs extends React.Component {
    static defaultProps = { tabs: []};
    state = {
        currentTabIndex: 0
    }
    handleButtonClick(index){
        this.setState({ currentTabIndex: index})
    }
    render(){
        const buttons = this.props.tabs.map((tab, index) => (
            <button key={index} onClick={() => this.handleButtonClick(index)}>
                {tab.name}
            </button>
        ))
        const currentTab = this.props.tabs[this.state.currentTabIndex]
        return(
            <div>
                {buttons}
                {this.props.tabs.length && (
                <div>
                    {currentTab.content}
                </div>
                )}
            </div>
        )
    }
}

export default Tabs