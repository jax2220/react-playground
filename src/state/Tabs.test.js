import React from 'react';
import ReactDOM from 'react-dom';
import renderer  from 'react-test-renderer'
import Tabs from './Tabs';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Tabs component', () => {
    const tabsProp = [
        {name: 'First tab',
         content: 'Lorem ipsum dolor sit amet consectetur'},
        {name: 'Second tab',
         content: 'Laboriosam exercitationem quos consectetur '},
        {name: 'Third tab',
         content: 'Fugit, sapiente aspernatur corporis velit'},
      ];

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Tabs />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the first tab by default', () => {
        const tree = renderer.create(<Tabs tabs={tabsProp} />)
        expect(tree).toMatchSnapshot()
    })

    it('renders impty given no tabs', () => {
        const wrapper = shallow(<Tabs />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

