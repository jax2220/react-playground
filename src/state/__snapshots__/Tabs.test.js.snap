// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders impty given no tabs 1`] = `
<div>
  0
</div>
`;

exports[`Tabs component renders the first tab by default 1`] = `
<div>
  <button>
    First tab
  </button>
  <button>
    Second tab
  </button>
  <button>
    Third tab
  </button>
  <div>
    Lorem ipsum dolor sit amet consectetur
  </div>
</div>
`;
